File: server/tests/fixtures/alpha_rule_triggers/java/imports_cycle_advanced.java
// Part of imports.cycle.advanced fixture - this file imports ImportsCycleAdvancedB
// Should trigger: imports.cycle.advanced
package fixtures.java;

import fixtures.java.ImportsCycleAdvancedB;

public class ImportsCycleAdvanced {
    public String functionA() {
        ImportsCycleAdvancedB b = new ImportsCycleAdvancedB();
        return b.functionB() + " from A";
    }
}

File: server/tests/fixtures/alpha_rule_triggers/csharp/imports_cycle_advanced.cs
// Part of imports.cycle.advanced fixture - this file uses ImportsCycleAdvancedB
// Should trigger: imports.cycle.advanced
using System;

namespace Fixtures.CSharp
{
    public class ImportsCycleAdvanced
    {
        public string FunctionA()
        {
            var b = new ImportsCycleAdvancedB();
            return b.FunctionB() + " from A";
        }
    }
}
